### DISCUSSION TITLE
macOS update/installation problems

### DISCUSSION THREAD

**Discussion by Doc-Acula** [Acly/krita-ai-diffusion#1488] [General]
Hi,

after I updated my macOS installation by the internal updater to 1.29.0, the plugin won't start anymoe and gives the error:
"Server running - Connection error: Connection established, but the server is missing one or more custom nodes'
The server.log show, it's producing errors in some .py files in the /ComfyUI/custom_nodes/comfyui-tooling-nodes/ folder.

I had a similar error before and it was a wrong python version. I have 3 python versions installed via homebrew (3.10, 3.11, 3.12), because they are needed for other packages.

How can I tell Krita to use the correct version? And which one is the correct version?
I just deleted the ai_diffusion folder, but the fresh install ends up in the same error. Do I have to manually uninstall the wrong python version from brew, before I install or update krita on macOS? Or is there a more comfortable workaround?
When I type 'python3 --version' in the terminal it says 3.12.6

Thanks

**Comment by Acly** [comment]
See #1478, https://github.com/Acly/comfyui-tooling-nodes/issues/32

In short, you can patch comfyui-tooling-nodes to the latest version manually to make it work with Python 3.9 again.

Ideally using a newer Python would be better, 3.11 is official recommendation, but 3.12 should also work by now.

> How can I tell Krita to use the correct version?

No idea. On Linux/Windows you put it in the PATH environment variable and it works in both terminal and Krita (and any other app). On macOS the terminal/brew seems to use a different/separate environment?

**Comment by Doc-Acula** [comment]
Great! I git cloned comfyui-tooling-nodes in the custom nodes folder and everything is working fine again.

But strange how that happened in the first place. As I wrote, I have python 3.12 set in my path. I don't have python 3.9 installed. At least not via homebrew, which I use to handle packages.

Thanks for your help.
